src/azure-cli/azure/cli/command_modules/network/_validators.py

```diff
@@ -1911,3 +1911,23 @@ def process_private_endpoint_connection_id_argument(cmd, namespace):
     namespace.resource_provider = '{}/{}'.format(result['namespace'], result['type'])
     namespace.name = result['child_name_1']
     del namespace.connection_id
+
+
+def process_vnet_name_or_id(cmd, namespace):
+    from msrestazure.tools import is_valid_resource_id, resource_id
```

Comment:
use is_valid_resource_id, resource_id from azure.mgmt.core.tools instead of msrest.